From b4c44da9c355cf8ce7aa49dd14b7a3805f1f9779 Mon Sep 17 00:00:00 2001
From: Christopher Obbard <chris.obbard@collabora.com>
Date: Tue, 4 Feb 2020 13:12:55 +0000
Subject: Pass system version to bundle install hooks

Pass installed system version to bundle install-check hook script.

Signed-off-by: Christopher Obbard <chris.obbard@collabora.com>

Gbp-Pq: Topic atari
Gbp-Pq: Name pass-system-version-to-bundle-install-hooks.patch
---
 src/install.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/install.c b/src/install.c
index 3e5e44d..dc52e64 100644
--- a/src/install.c
+++ b/src/install.c
@@ -601,6 +601,8 @@ static gboolean run_bundle_hook(RaucManifest *manifest, gchar* bundledir, const
 	g_autoptr(GDataInputStream) datainstream = NULL;
 	gboolean res = FALSE;
 	gchar *outline, *hookreturnmsg = NULL;
+	g_autofree gchar *system_version = NULL;
+	RaucSlot *s = NULL;
 
 	g_assert_nonnull(manifest->hook_name);
 
@@ -608,10 +610,21 @@ static gboolean run_bundle_hook(RaucManifest *manifest, gchar* bundledir, const
 
 	g_message("Running bundle hook %s", hook_cmd);
 
+	/* pass the version of the current slot to the hooks script */
+	/* a proper method is TBD upstream, see: https://github.com/rauc/rauc/issues/531 */
+	s = find_config_slot_by_device(r_context()->config, r_context()->bootslot);
+	if (s) {
+		load_slot_status(s);
+		if (s->status)
+			system_version = g_strdup(s->status->bundle_version);
+	}
+	g_message("bootslot=%s, system_version=%s", r_context()->bootslot, system_version ?: "");
+
 	launcher = g_subprocess_launcher_new(G_SUBPROCESS_FLAGS_STDERR_PIPE);
 
 	g_subprocess_launcher_setenv(launcher, "RAUC_SYSTEM_COMPATIBLE", r_context()->config->system_compatible, TRUE);
 	g_subprocess_launcher_setenv(launcher, "RAUC_SYSTEM_VARIANT", r_context()->config->system_variant ?: "", TRUE);
+	g_subprocess_launcher_setenv(launcher, "RAUC_SYSTEM_VERSION", system_version ?: "", TRUE);
 
 	g_subprocess_launcher_setenv(launcher, "RAUC_MF_COMPATIBLE", manifest->update_compatible, TRUE);
 	g_subprocess_launcher_setenv(launcher, "RAUC_MF_VERSION", manifest->update_version ?: "", TRUE);
